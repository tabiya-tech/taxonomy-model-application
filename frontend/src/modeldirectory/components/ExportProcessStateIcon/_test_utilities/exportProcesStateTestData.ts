import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllExportProcessStatePermutations = (): ModelInfoTypes.ExportProcessState[] => {
  // generate all permutations of ExportProcessState status, errored, exportErrors, exportWarnings
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ExportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((exportErrors) => {
        allBooleans.forEach((exportWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              exportErrors: exportErrors,
              exportWarnings: exportWarnings,
            },
            downloadUrl: "https://example.com",
            timestamp: new Date(),
          });
        });
      });
    });
  });
  return allPermutations;
};
