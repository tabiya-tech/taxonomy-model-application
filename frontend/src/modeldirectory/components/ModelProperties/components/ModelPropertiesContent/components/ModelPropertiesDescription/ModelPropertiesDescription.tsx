import React from "react";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import Box from "@mui/material/Box";
import { useTheme } from "@mui/material";
import TextPropertyField from "src/theme/PropertyFieldLayout/TextPropertyField/TextPropertyField";
import MarkdownPropertyField from "src/theme/PropertyFieldLayout/MarkdownPropertyField/MarkdownPropertyField";

export interface ModelPropertiesDescriptionProps {
  model: ModelInfoTypes.ModelInfo;
}

const uniqueId = "2dc9c69b-bd8f-4b51-bff8-69faa511a316";

export const DATA_TEST_ID = {
  MODEL_PROPERTIES_DESCRIPTION_CONTAINER: `model-properties-description-container-${uniqueId}`,
  MODEL_PROPERTIES_NAME: `model-properties-name-${uniqueId}`,
  MODEL_PROPERTIES_LOCALE: `model-properties-locale-${uniqueId}`,
  MODEL_PROPERTIES_DESCRIPTION: `model-properties-description-${uniqueId}`,
};

export const FIELD_ID = {
  NAME: `name-${uniqueId}`,
  LOCALE: `locale-${uniqueId}`,
  DESCRIPTION: `description-${uniqueId}`,
};

export const FIELD_LABEL_TEXT = {
  LABEL_NAME: "Name",
  LABEL_LOCALE: "Locale",
  LABEL_DESCRIPTION: "Description",
};

/**
 * ModelPropertiesDescription responsible for displaying the basic details of a model
 * like description, locale, etc.
 * @param props
 * @constructor
 */
const ModelPropertiesDescription: React.FC<ModelPropertiesDescriptionProps> = (
  props: Readonly<ModelPropertiesDescriptionProps>
) => {
  const theme = useTheme();

  return (
    <Box
      display="flex"
      flexDirection="column"
      gap={theme.tabiyaSpacing.md}
      data-testid={DATA_TEST_ID.MODEL_PROPERTIES_DESCRIPTION_CONTAINER}
    >
      <TextPropertyField
        label={FIELD_LABEL_TEXT.LABEL_NAME}
        text={`${props.model.name}`}
        data-testid={DATA_TEST_ID.MODEL_PROPERTIES_NAME}
        fieldId={FIELD_ID.NAME}
      />
      <TextPropertyField
        label={FIELD_LABEL_TEXT.LABEL_LOCALE}
        text={`${props.model.locale.name}(${props.model.locale.shortCode})`}
        data-testid={DATA_TEST_ID.MODEL_PROPERTIES_LOCALE}
        fieldId={FIELD_ID.LOCALE}
      />
      <MarkdownPropertyField
        label={FIELD_LABEL_TEXT.LABEL_DESCRIPTION}
        text={props.model.description}
        data-testid={DATA_TEST_ID.MODEL_PROPERTIES_DESCRIPTION}
        fieldId={FIELD_ID.DESCRIPTION}
      />
    </Box>
  );
};

export default ModelPropertiesDescription;
