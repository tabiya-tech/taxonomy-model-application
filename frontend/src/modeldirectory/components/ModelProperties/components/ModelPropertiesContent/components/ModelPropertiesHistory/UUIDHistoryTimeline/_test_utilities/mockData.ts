import { ModelInfoTypes } from "src//modelInfo/modelInfoTypes";

export function getOneFakeUUIDHistoryDetail(i: number): ModelInfoTypes.ModelHistory {
  return {
    UUID: "e26aca13-aa00-4346-9e83-edaf59314378" + i,
    id: "5e12f537f079fbdcb4e61bf" + i,
    localeShortCode: "ZA",
    name: "Model" + i,
    version: "V1.1." + i,
  };
}

export function getFakeUUIDHistoryDetailsArray(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => getOneFakeUUIDHistoryDetail(i));
}

export function getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => {
    const details = getOneFakeUUIDHistoryDetail(i);
    details.id = null;
    details.name = null;
    details.version = null;
    details.localeShortCode = null;
    return details;
  });
}
