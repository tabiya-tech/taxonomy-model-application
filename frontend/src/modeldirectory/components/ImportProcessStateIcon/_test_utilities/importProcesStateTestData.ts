import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllImportProcessStatePermutations = (): ModelInfoTypes.ImportProcessState[] => {
  // generate all permutations of ImportProcessState status, errored, parsingErrors, parsingWarnings
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ImportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((parsingErrors) => {
        allBooleans.forEach((parsingWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              parsingErrors: parsingErrors,
              parsingWarnings: parsingWarnings,
            },
            createdAt: new Date(allPermutations.length),
            updatedAt: new Date(allPermutations.length),
          });
        });
      });
    });
  });
  return allPermutations;
};
