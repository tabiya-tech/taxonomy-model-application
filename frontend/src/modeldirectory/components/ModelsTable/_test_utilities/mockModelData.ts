import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import ExportProcessStateAPISpecs from "api-specifications/exportProcessState";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";

import { v4 as uuidv4 } from "uuid";

import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getMockId } from "src/_test_utilities/mockMongoId";
import { getRandomLorem, getRandomString, getTestString } from "src/_test_utilities/specialCharacters";
import { faker } from "@faker-js/faker";
import { CELL_MAX_LENGTH } from "src/modeldirectory/components/ModelsTable/ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  const model = getArrayOfFakeModels(1)[0];
  model.id = getMockId(id ?? 1);
  return model;
}

export function getArrayOfFakeModels(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location
          .countryCode("alpha-3")
          .substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description:
        i % 2 === 0
          ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH)
          : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getArrayOfFakeModelsMaxLength(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export const fakeModel: ModelInfoTypes.ModelInfo = {
  UUID: "8d914eab-6f7d-4183-accc-a09f99887b39",
  UUIDHistory: ["8d914eab-6f7d-4183-accc-a09f99887b39"],
  description: "aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm",
  id: "000000000000000000000001",
  importProcessState: {
    id: "000000000000000000000001",
    result: {
      errored: true,
      parsingErrors: true,
      parsingWarnings: true,
    },
    status: ImportProcessStateEnums.Status.PENDING,
    createdAt: new Date("2023-10-18T17:35:10.571Z"),
    updatedAt: new Date("2023-10-18T17:35:12.571Z"),
  },
  exportProcessState: [
    {
      downloadUrl: "https://www.example.com/",
      id: "000000000000000000000001",
      result: {
        errored: true,
        exportErrors: true,
        exportWarnings: true,
      },
      status: ExportProcessStateAPISpecs.Enums.Status.PENDING,
      timestamp: new Date("2023-10-18T17:35:10.571Z"),
      createdAt: new Date("2023-10-18T17:35:10.571Z"),
      updatedAt: new Date("2023-10-18T17:35:12.571Z"),
    },
  ],
  locale: {
    UUID: "7c4ea00b-fbea-4a31-9c04-009c813f5f34",
    name: "Palestine",
    shortCode: "ETH",
  },
  name: "1/1 - fysmros tcgcnjbbrrev",
  path: "https://unwelcome-editorial.net/",
  releaseNotes: "Perferendis modi impedit necessitatibus a",
  released: true,
  tabiyaPath: "https://silver-vulture.biz",
  createdAt: new Date("2023-10-18T17:35:10.571Z"),
  updatedAt: new Date("2022-10-18T17:35:10.571Z"),
  version: "0.2.2",
};

export function getRandomImportStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getRandomExportStatus(id: number) {
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getOneFakeExportProcessState(i: number): ModelInfoTypes.ExportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomExportStatus(i),
    result: {
      errored: i % 2 === 0,
      exportErrors: i % 2 === 0,
      exportWarnings: i % 2 === 0,
    },
    downloadUrl: faker.internet.url(),
    timestamp: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}

export function getOneFakeImportProcessState(i: number): ModelInfoTypes.ImportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomImportStatus(i),
    result: {
      errored: i % 2 === 0,
      parsingErrors: i % 2 === 0,
      parsingWarnings: i % 2 === 0,
    },
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}
