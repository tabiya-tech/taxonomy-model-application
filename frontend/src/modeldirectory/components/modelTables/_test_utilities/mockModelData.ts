import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import {v4 as uuidv4} from "uuid";

import {ModelInfoTypes} from "src/modelInfo/modelInfoTypes";
import {getMockId} from "src/_test_utilities/mockMongoId";
import {getRandomLorem, getRandomString, getTestString} from "src/_test_utilities/specialCharacters";
import {faker} from "@faker-js/faker";
import {CELL_MAX_LENGTH} from "../ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  return getArrayOfFakeModels(id ? id : 1)[0];
}

export function getArrayOfFakeModels(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: Array.from({length: 20}, (_, i) => faker.lorem.word()).join(" ").substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location.countryCode("alpha-3").substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: i % 2 === 0 ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH) : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getArrayOfFakeModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      importProcessState: {
        id: getMockId(10000 + i),
        status: getRandomStatus(i),
        result: {
          errored: i % 2 === 0,
          parsingErrors: i % 2 === 0,
          parsingWarnings: i % 2 === 0,
        }
      }
    };
    models.push(model);
  }
  return models
}

export function getArrayOfFakeModelsForSorting(): ModelInfoTypes.ModelInfo[] {
  const baseModel: Omit<ModelInfoTypes.ModelInfo, 'name' | 'version' | 'createdAt' | 'updatedAt'> = {
    id: 'base',
    UUID: uuidv4(),
    previousUUID: uuidv4(),
    originUUID: uuidv4(),
    locale: {
      UUID: uuidv4(),
      name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      shortCode: faker.location.countryCode("alpha-3").substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
    },
    description: getRandomLorem(CELL_MAX_LENGTH / 2),
    released: true,
    releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
    path: faker.internet.url(),
    tabiyaPath: faker.internet.url(),
    importProcessState: {
      id: getMockId(10000),
      status: getRandomStatus(1),
      result: {
        errored: true,
        parsingErrors: true,
        parsingWarnings: true,
      }
    }
  };

  return [
    {
      ...baseModel,
      id: '1',
      name: 'ancient',
      version: '1.0.0',
      createdAt: new Date('2022-01-01T00:00:00.000Z'),
      updatedAt: new Date('2022-01-01T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '2',
      name: 'older',
      version: '1.1.0',
      createdAt: new Date('2022-01-02T00:00:00.000Z'),
      updatedAt: new Date('2022-01-02T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '3',
      name: 'old',
      version: '1.1.1',
      createdAt: new Date('2022-01-03T00:00:00.000Z'),
      updatedAt: new Date('2022-01-03T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '4',
      name: 'new',
      version: '1.1.2',
      createdAt: new Date('2022-01-04T00:00:00.000Z'),
      updatedAt: new Date('2022-01-04T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '5',
      name: 'newer',
      version: '1.1.3',
      createdAt: new Date('2022-01-05T00:00:00.000Z'),
      updatedAt: new Date('2022-01-05T00:00:00.000Z'),
    },
    {
      ...baseModel,
      id: '6',
      name: 'newest',
      version: '1.2.2',
      createdAt: new Date('2022-01-06T00:00:00.000Z'),
      updatedAt: new Date('2022-01-06T00:00:00.000Z'),
    }
  ];
}


export function getRandomStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}
