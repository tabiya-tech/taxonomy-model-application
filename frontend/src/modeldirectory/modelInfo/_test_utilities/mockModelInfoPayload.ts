import { faker } from '@faker-js/faker';

import * as ModelInfo from "api-specifications/modelInfo";
import {v4 as uuidv4} from "uuid";

import {getMockId} from "src/_test_utilities/mockMongoId";
import {getRandomString, getTestString} from "src/_test_utilities/specialCharacters";

export function getMockModelInfoPayload(number: number): ModelInfo.Types.GET.Response.Payload {
  return Array.from({length: number}, (_, i) => {
    return {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: getRandomString(ModelInfo.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(ModelInfo.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(ModelInfo.Constants.LOCALE_SHORTCODE_MAX_LENGTH)
      },
      description: getTestString(ModelInfo.Constants.DESCRIPTION_MAX_LENGTH),
      released: false,
      releaseNotes: getTestString(ModelInfo.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfo.Constants.VERSION_MAX_LENGTH),
      // @ts-ignore
      createdAt: new Date().toISOString(),
      // @ts-ignore
      updatedAt: new Date().toISOString(),
      path: "https://foo/bar",
      tabiyaPath: "https://foo/bar/baz"
    }
  });
}

export function getFakeModelInfoPayload(number: number): ModelInfo.Types.GET.Response.Payload {
  return Array.from({length: number}, (_, i) => {
    return {
      id: getMockId(i),
      UUID: uuidv4(),
      previousUUID: uuidv4(),
      originUUID: uuidv4(),
      name: faker.lorem.text().substring(0, ModelInfo.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country(),
        shortCode: faker.lorem.text().substring(0,  ModelInfo.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: faker.lorem.text().substring(0, ModelInfo.Constants.DESCRIPTION_MAX_LENGTH),
      released: false,
      releaseNotes: faker.lorem.text().substring(0, ModelInfo.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.lorem.text().substring(0, ModelInfo.Constants.VERSION_MAX_LENGTH),
      // @ts-ignore
      createdAt: faker.date.anytime().toISOString(),
      // @ts-ignore
      updatedAt: faker.date.anytime().toISOString(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url()
    }
  });
}