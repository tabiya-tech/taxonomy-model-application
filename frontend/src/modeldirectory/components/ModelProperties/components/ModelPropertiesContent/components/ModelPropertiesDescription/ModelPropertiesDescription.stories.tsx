import { getOneFakeModel } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import { Meta, StoryObj } from "@storybook/react";
import ModelPropertiesDescription from "./ModelPropertiesDescription";

const meta: Meta<typeof ModelPropertiesDescription> = {
  title: "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesDescription",
  component: ModelPropertiesDescription,
  tags: ["autodocs"],
  args: {
    model: getOneFakeModel(1),
  },
};

export default meta;

type Story = StoryObj<typeof ModelPropertiesDescription>;

export const Shown: Story = {};
