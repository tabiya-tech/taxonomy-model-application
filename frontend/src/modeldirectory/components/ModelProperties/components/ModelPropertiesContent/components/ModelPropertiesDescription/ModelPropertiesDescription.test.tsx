// mute the console
import "src/_test_utilities/consoleMock";

import { fakeModel } from "src/modeldirectory/components/ModelsTable/_test_utilities/mockModelData";
import { render, screen } from "src/_test_utilities/test-utils";
import ModelPropertiesDescription, { DATA_TEST_ID, FIELD_ID, FIELD_LABEL_TEXT } from "./ModelPropertiesDescription";
import TextPropertyField from "src/theme/PropertyFieldLayout/TextPropertyField/TextPropertyField";

// mock the TextPropertyField component
jest.mock("src/theme/PropertyFieldLayout/TextPropertyField/TextPropertyField", () => {
  return {
    __esModule: true,
    default: jest.fn().mockImplementation((props) => (
      <div data-testid={props["data-testid"]} id={props.fieldId}>
        Text Property Field Mock
      </div>
    )),
  };
});

describe("ModelPropertiesDescription", () => {
  test("should render correctly with the provided model props", () => {
    // GIVEN a model
    const givenModel = fakeModel;

    // WHEN the ModelPropertiesDescription is rendered with the given model
    render(<ModelPropertiesDescription model={givenModel} />);

    // THEN expect no errors or warning to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the component to be shown
    const modelPropertiesDescriptionContainer = screen.getByTestId(DATA_TEST_ID.MODEL_PROPERTIES_DESCRIPTION_CONTAINER);
    expect(modelPropertiesDescriptionContainer).toBeInTheDocument();
    // AND the name property to be shown
    const nameComponent = screen.getByTestId(DATA_TEST_ID.MODEL_PROPERTIES_NAME);
    expect(nameComponent).toBeInTheDocument();
    // AND the TextPropertyField component to be called with the correct props for the 'Name'
    expect(TextPropertyField).toHaveBeenCalledWith(
      {
        label: FIELD_LABEL_TEXT.LABEL_NAME,
        text: givenModel.name,
        "data-testid": DATA_TEST_ID.MODEL_PROPERTIES_NAME,
        fieldId: FIELD_ID.NAME,
      },
      {}
    );
    // AND locale property to be shown
    const localeComponent = screen.getByTestId(DATA_TEST_ID.MODEL_PROPERTIES_LOCALE);
    expect(localeComponent).toBeInTheDocument();
    // AND the TextPropertyField component to be called with the correct props for the 'Locale'
    expect(TextPropertyField).toHaveBeenCalledWith(
      {
        label: FIELD_LABEL_TEXT.LABEL_LOCALE,
        text: `${givenModel.locale.name}(${givenModel.locale.shortCode})`,
        "data-testid": DATA_TEST_ID.MODEL_PROPERTIES_LOCALE,
        fieldId: FIELD_ID.LOCALE,
      },
      {}
    );
    // AND the description property to be shown
    const descriptionItem = screen.getByTestId(DATA_TEST_ID.MODEL_PROPERTIES_DESCRIPTION);
    expect(descriptionItem).toBeInTheDocument();
    // AND the TextPropertyField component to be called with the correct props for the 'Description'
    expect(TextPropertyField).toHaveBeenCalledWith(
      {
        label: FIELD_LABEL_TEXT.LABEL_DESCRIPTION,
        text: givenModel.description,
        "data-testid": DATA_TEST_ID.MODEL_PROPERTIES_DESCRIPTION,
        fieldId: FIELD_ID.DESCRIPTION,
      },
      {}
    );
    // AND to match the snapshot
    expect(modelPropertiesDescriptionContainer).toMatchSnapshot();
  });
});
