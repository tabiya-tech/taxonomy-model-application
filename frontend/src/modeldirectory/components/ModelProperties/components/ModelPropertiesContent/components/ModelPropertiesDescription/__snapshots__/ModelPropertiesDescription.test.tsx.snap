// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelPropertiesDescription should render correctly with the provided model props 1`] = `
<div
  class="MuiBox-root css-1ck9pm7"
  data-testid="model-properties-description-container-2dc9c69b-bd8f-4b51-bff8-69faa511a316"
>
  <div
    data-testid="model-properties-name-2dc9c69b-bd8f-4b51-bff8-69faa511a316"
  >
    <span>
      Name
    </span>
    <span>
      1/1 - fysmros tcgcnjbbrrev
    </span>
  </div>
  <div
    data-testid="model-properties-locale-2dc9c69b-bd8f-4b51-bff8-69faa511a316"
  >
    <span>
      Locale
    </span>
    <span>
      Palestine(ETH)
    </span>
  </div>
  <div
    data-testid="model-properties-description-2dc9c69b-bd8f-4b51-bff8-69faa511a316"
  >
    <span>
      Description
    </span>
    <span>
      aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm
    </span>
  </div>
</div>
`;
